waku/waku_lightpush/protocol.nim

```diff
@@ -26,92 +26,90 @@ type WakuLightPush* = ref object of LPProtocol
+    error "lightpush request handling error", error = msg
+    return
+      WakuLightPushResult.err((code: LightpushStatusCode.BAD_REQUEST, desc: some(msg)))
+
+  waku_lightpush_v3_messages.inc(labelValues = ["PushRequest"])
+
+  let msg_hash = pubsubTopic.computeMessageHash(pushRequest.message).to0xHex()
+  notice "handling lightpush request",
+    my_peer_id = wl.peerManager.switch.peerInfo.peerId,
+    peer_id = peerId,
+    requestId = pushRequest.requestId,
+    pubsubTopic = pushRequest.pubsubTopic,
+    msg_hash = msg_hash,
+    receivedTime = getNowInNanosecondTime()
+
+  let handleRes = await wl.pushHandler(peerId, pubsubTopic, pushRequest.message)
+
+  if handleRes.isOk():
+    return ok(handleRes.get())
+
+  return err((code: handleRes.error.code, desc: handleRes.error.desc))
 
 proc handleRequest*(
     wl: WakuLightPush, peerId: PeerId, buffer: seq[byte]
 ): Future[LightPushResponse] {.async.} =
-  let reqDecodeRes = LightpushRequest.decode(buffer)
-  var isSuccess = false
   var pushResponse: LightpushResponse
 
-  if reqDecodeRes.isErr():
-    pushResponse = LightpushResponse(
+  let pushRequest = LightpushRequest.decode(buffer).valueOr:
+    let desc = decodeRpcFailure & ": " & $error
+    waku_lightpush_v3_errors.inc(labelValues = [desc])
+    error "failed to push message", error = pushResponse.statusDesc
+    return LightpushResponse(
       requestId: "N/A", # due to decode failure we don't know requestId
       statusCode: LightpushStatusCode.BAD_REQUEST.uint32,
-      statusDesc: some(decodeRpcFailure & ": " & $reqDecodeRes.error),
+      statusDesc: some(desc),
     )
-  else:
-    let pushRequest = reqDecodeRes.get()
-
-    let pubsubTopic = pushRequest.pubSubTopic.valueOr:
-      let parsedTopic = NsContentTopic.parse(pushRequest.message.contentTopic).valueOr:
-        let msg = "Invalid content-topic:" & $error
-        error "lightpush request handling error", error = msg
-        return LightpushResponse(
-          requestId: pushRequest.requestId,
-          statusCode: LightpushStatusCode.INVALID_MESSAGE_ERROR.uint32,
-          statusDesc: some(msg),
-        )
-
-      wl.sharding.getShard(parsedTopic).valueOr:
-        let msg = "Autosharding error: " & error
-        error "lightpush request handling error", error = msg
-        return LightpushResponse(
-          requestId: pushRequest.requestId,
-          statusCode: LightpushStatusCode.INTERNAL_SERVER_ERROR.uint32,
-          statusDesc: some(msg),
-        )
 
-    # ensure checking topic will not cause error at gossipsub level
-    if pubsubTopic.isEmptyOrWhitespace():
-      let msg = "topic must not be empty"
-      error "lightpush request handling error", error = msg
-      return LightPushResponse(
-        requestId: pushRequest.requestId,
-        statusCode: LightpushStatusCode.BAD_REQUEST.uint32,
-        statusDesc: some(msg),
-      )
+  # TODO: is that fine?
+  let res = await handleRequest(wl, peerId, pushRequest)
```

Comment:
@NagyZoltanPeter 

Is that ok to call an await here? I am not yet familiar with Nim async and whether there are any gotcha